test/test-genesis.ts

```diff
@@ -31,66 +53,108 @@ describe("Genesis Contract", () => {
   });
 
   // TODO Adjust with real values
-  it("Should initialize the Genesis contract", async () => {
+  it('Should initialize the Genesis contract', async () => {
     expect(await contract.MAX_SUPPLY()).to.equal(1000);
-    expect(await contract.PRICE()).to.equal(utils.parseEther("0.0000001"));
+    expect(await contract.PRICE()).to.equal(utils.parseEther('0.0000001'));
     expect(await contract.MAX_PER_MINT()).to.equal(1);
-    expect(await contract.presaleActive()).to.be.false;
     expect(await contract.mintActive()).to.be.false;
-    expect(await contract.reservesMinted()).to.be.false;
   });
 
-  it("Should set the right owner", async () => {
+  it('Should set the right owner', async () => {
     expect(await contract.owner()).to.equal(await owner.address);
   });
 
-  it("Should not allow minting if it is not active", async function () {
-    await expect(contract.connect(address1).mint()).to.be.revertedWith(
-      "Minting is not active yet!",
-    );
-    await expect(contract.connect(owner).mintNFTs(5)).to.be.revertedWith(
-      "Minting is not active yet!",
-    );
+  it('Should not allow minting if it is not active', async function () {
+    await expect(
+      mint(
+        owner,
+        bogusNonce,
+        bogusProof,
+        contract,
+        oracle,
+        VRFCoordinatorMock.address,
+      ),
+    ).to.be.revertedWith('Minting is not active yet!');
   });
 
-  it("Should allow minting if it is active", async function () {
+  it('Should allow minting if it is active', async function () {
     await contract.connect(owner).flipMintActive();
     expect(await contract.mintActive()).to.be.true;
+    await expect(
+      mint(
+        owner,
+        bogusNonce,
+        bogusProof,
```

Comment:
Why can he mint with a `bogusNonce` and `bogusProof`?

Reply:
because owner bypass WL check. Will be removed if we can set the address

Reply:
yea, this needs to be figured out, check `smock` because it leads to weird tests IMO.